doc/api/replication.md

```diff
@@ -0,0 +1,106 @@
+# Thali Cordova Peer Communication API #
+
```

Comment:
It would be good to have some kind of introduction to this section. We should probably explain that people who are using the full Thali stack don't need to know about this API but that people who want to re-use our discovery and p2p connectivity functionality can use this API to enable their own TCP/IP based servers to be discoverable and available over p2p in a device independent way.